### PR TITLE
Add the handler of open video error in action_recognition demo

### PR DESCRIPTION
**Pull request by anzhella-pankratova**
There was no handling of situation when the demo fails to open the input video (one or many). Add the handler of this error to prevent the demo further running.

### CODE REVIEW

**Comment by vladimir-dudnik** [comment]
@anzhella-pankratova please see CI errors:
`16:24:13  Traceback (most recent call last):
16:24:13    File "/home/jenkins/workspace/omz/precommit/demos_ubuntu18@2/src/demos/python_demos/action_recognition/action_recognition.py", line 27, in <module>
16:24:13      from action_recognition_demo.steps import run_pipeline
16:24:13    File "/home/jenkins/workspace/omz/precommit/demos_ubuntu18@2/src/demos/python_demos/action_recognition/action_recognition_demo/steps.py", line 60
16:24:13      self._open_video()
16:24:13                       ^
16:24:13  TabError: inconsistent use of tabs and spaces in indentation`

**Comment by vladimir-dudnik** [comment]
@anzhella-pankratova please fix pre-commit check issues:
13:42:08  + ./venv/bin/python ./src/ci/check-basics.py
13:42:13  demos/python_demos/action_recognition/action_recognition.py:129: last line doesn't end with a newline character
13:42:13  demos/python_demos/action_recognition/action_recognition_demo/pipeline.py:182: last line doesn't end with a newline character
13:42:13  demos/python_demos/action_recognition/action_recognition_demo/steps.py:180: last line doesn't end with a newline character

File: demos/python_demos/action_recognition/action_recognition.py [hunk 1]
```diff
@@ -69,18 +68,12 @@ def build_argparser():
...
-    else:
-        videos = [args.input]
-
-    if not args.input:
+    input_path = args.input
+    if not input_path:
         raise ValueError('--input option is expected')
+    if not path.isfile(input_path):
+        raise ValueError('The input file does not exists')
+    video = [input_path]
```
Comment (Wovchena) [inline demos/python_demos/action_recognition/action_recognition.py:76]: We don't need any of this. VideoCapture fails when it encounters empty string. And non-file inputs are valid. For example `0` for camera.

So run_pipeline() could just take `args.input`.

File: demos/python_demos/action_recognition/action_recognition_demo/steps.py [hunk 2]
```diff
@@ -27,9 +27,9 @@
 from .queue import Signal
 
 
-def run_pipeline(video, encoder, decoder, render_fn, decoder_seq_size=16, fps=30):
+def run_pipeline(video, encoder, decoder, render_fn, loop, decoder_seq_size=16, fps=30):
```
Comment (Wovchena) [inline demos/python_demos/action_recognition/action_recognition_demo/steps.py:30]: Put `loop` arg after `video`. They are linked.

File: demos/python_demos/action_recognition/action_recognition_demo/steps.py [hunk 3]
```diff
@@ -41,25 +41,37 @@ def run_pipeline(video, encoder, decoder, render_fn, decoder_seq_size=16, fps=30
 
 class DataStep(PipelineStep):
 
-    def __init__(self, video_list, loop=True):
+    def __init__(self, video_filepath, loop):
```
Comment (Wovchena) [inline demos/python_demos/action_recognition/action_recognition_demo/steps.py:44]: Rename `video_filepath` to `input` to be general enough.

File: demos/python_demos/action_recognition/action_recognition_demo/steps.py [hunk 4]
```diff
@@ -41,25 +41,37 @@ def run_pipeline(video, encoder, decoder, render_fn, decoder_seq_size=16, fps=30
...
+        if self.loop:
+            self._video_cycle = cycle(self.video)
         else:
-            self._video_cycle = iter(self.video_list)
+            self._video_cycle = iter(self.video)
+
+        try:
+            self._open_video()
+            if not self.loop:
+                self._video_cycle = iter(self.video)
```
Comment (Wovchena) [inline demos/python_demos/action_recognition/action_recognition_demo/steps.py:58]: You should open the video only once. You won't need to iterate over `self.video`.

File: demos/python_demos/action_recognition/action_recognition_demo/steps.py [hunk 5]
```diff
@@ -41,25 +41,37 @@ def run_pipeline(video, encoder, decoder, render_fn, decoder_seq_size=16, fps=30
...
+        except:
+            print("Error: The input video cannot be opened")
+            exit(1)
```
Comment (Wovchena) [inline demos/python_demos/action_recognition/action_recognition_demo/steps.py:61]: There is no need to use both approaches. Either `exit` right from `_open_video()` or just let the exception break the whole demo.
You can even remove `_open_video()` and put its body into init.

File: demos/python_demos/action_recognition/action_recognition_demo/steps.py [hunk 6]
```diff
@@ -41,25 +41,37 @@ def run_pipeline(video, encoder, decoder, render_fn, decoder_seq_size=16, fps=30
...
 
     def setup(self):
         self._open_video()
```
Comment (Wovchena) [inline demos/python_demos/action_recognition/action_recognition_demo/steps.py:64]: Make `setup()` empty.

File: demos/python_demos/action_recognition/action_recognition_demo/steps.py [hunk 7]
```diff
@@ -41,25 +41,37 @@ def run_pipeline(video, encoder, decoder, render_fn, decoder_seq_size=16, fps=30
...
 
     def process(self, item):
-        if not self.cap.isOpened() and not self._open_video():
-            return Signal.STOP
         status, frame = self.cap.read()
         if not status:
-            return Signal.STOP
+            if self.loop:
+                self._open_video()
```
Comment (Wovchena) [inline demos/python_demos/action_recognition/action_recognition_demo/steps.py:70]:
```suggestion
```

File: demos/python_demos/action_recognition/action_recognition_demo/steps.py [hunk 8]
```diff
@@ -41,25 +41,37 @@ def run_pipeline(video, encoder, decoder, render_fn, decoder_seq_size=16, fps=30
...
+                self.cap.set(cv2.CAP_PROP_POS_FRAMES, 0)
+                status, frame = self.cap.read()
```
Comment (Wovchena) [inline demos/python_demos/action_recognition/action_recognition_demo/steps.py:72]: Check that new `status` value again and stop if it's false.

**Review by Wovchena — CHANGES_REQUESTED** [review]
I think we will align help messages with other demos in scope of next task.

**Comment by anzhella-pankratova** [comment]
@Wovchena Could you check my last commit?

**Review by Wovchena — APPROVED** [review]

**Pull request merged**